nimterop/modules.nim

```diff
@@ -0,0 +1,31 @@
+##[
+Module that imports everything so that `nim doc --project` runs docs
+on everything.
+]##
+
+when true:
+  ## pending https://github.com/nim-lang/Nim/pull/10527
+  import sequtils, os, strformat, macros
+  # using "."/ would give similar error as D20190208T153915
+  import nimterop/[paths, compat]
+  macro importPaths(a: static openArray[string]): untyped =
+    result = newStmtList()
+    for ai in a: result.add quote do: from `ai` import nil
+
+  const dir = nimteropSrcDir()
+  const files = block:
+    var ret: seq[string]
+    for path in walkDirRec(dir, yieldFilter = {pcFile}):
+      if path.splitFile.ext != ".nim": continue
+      if path.splitFile.name in ["astold"]: continue
+      if path == currentSourcePath: continue
+      #[
+      note(D20190208T153915): using `relativePath` because some files in nimterop use `import nimterop/foo` and it'd otherwise give this error:
+Hint: tsgen [Processing]
+modules.nim(24, 15) template/generic instantiation of `importPaths` from here
+/Users/travis/.nimble/pkgs/nimterop-0.1.1/nimterop/cimport.nim(1, 2) Error: module names need to be unique per Nimble package; module clashes with /Users/travis/build/nimterop/nimterop/nimterop/cimport.nim
+      ]#
+      ret.add path.relativePath nimteropRoot()
+    ret
+  static: echo files
+  importPaths files
```

Comment:
Why is all.nim or api.nim not good enough? It is working. What is all this code for?

Reply:
its's more DRY, it imports all the source files (minus a blacklist, astold doesn't compile), so that when files are added/renamed/removed, this doesn't need to change

Reply:
I intend to factor this logic out into a separate reusable nimble package that nimterop can depend on, since it's a generally useful thing (document all modules, recursively, without manually specifying which ones); if you want me to revert to manual enumeration, I can do it

Reply:
all right, reverted in back to explicit enumeration for the sake of moving forward